src/tests/ftest/util/AgentUtils.py

```diff
@@ -127,31 +127,32 @@ def run_agent(basepath, server_list, client_list=None):
         sessions[client] = p
 
     # double check agent launched successfully
-    file_desc = sessions[client].stderr.fileno()
-    flags = fcntl.fcntl(file_desc, fcntl.F_GETFL)
-    fcntl.fcntl(file_desc, fcntl.F_SETFL, flags | os.O_NONBLOCK)
-    timeout = 5
-    start_time = time.time()
-    result = 0
-    pattern = "Starting daos_agent"
-    expected_data = ""
-    while True:
-        output = ""
-        try:
-            output = sessions[client].stderr.read()
-        except IOError as excpn:
-            if excpn.errno != errno.EAGAIN:
-                raise excpn
-            continue
-        match = re.findall(pattern, output)
-        expected_data += output
-        result += len(match)
-        if not output or time.time() - start_time > timeout:
-            print("<AGENT>: {}".format(expected_data))
-            raise AgentFailed("DAOS Agent didn't start!")
-        break
-    print("<AGENT> agent started and took %s seconds to start" % \
-            (time.time() - start_time))
+    for client in client_list:
+        file_desc = sessions[client].stderr.fileno()
+        flags = fcntl.fcntl(file_desc, fcntl.F_GETFL)
+        fcntl.fcntl(file_desc, fcntl.F_SETFL, flags | os.O_NONBLOCK)
+        timeout = 5
+        start_time = time.time()
+        result = 0
+        pattern = "Starting daos_agent"
+        expected_data = ""
+        while True:
+            output = ""
+            try:
+                output = sessions[client].stderr.read()
+            except IOError as excpn:
+                if excpn.errno != errno.EAGAIN:
+                    raise excpn
```

Comment:
Might I suggest wrapping this exception in an AgentFailed exception so callers of this function only have to catch one exception type.

Reply:
Now raises AgentFailed.